### ISSUE TITLE
twisted.test.test_amp.AMPTest.test_keyTooLong assumes argument name identity is preserved

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4486]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4486|
|Type|defect|
|Created|2010-06-04 13:16:24Z|
|Branch|https://github.com/twisted/twisted/tree/amp-key-identity-4486|

This test invokes a method using `**` to pass a value for a parameter with a very long name.  It then uses an identity check to verify that the string included in the exception is the same as the one passed to the original call.

Since this string passes through various layers of the interpreter, it doesn't seem reasonable to require that identity is preserved.  Instead, equality should be checked.

<details><summary>Searchable metadata</summary>

```
trac-id__4486 4486
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_amp_key_identity_4486 branches/amp-key-identity-4486
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__tests tests
time__1275657384000000 1275657384000000
changetime__1275796386000000 1275796386000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [29181]) Branching to 'amp-key-identity-4486'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [29182]) Change an identity check to an equality check; also update a bunch of non-preferred aliases to use the preferred names

refs [#4486](https://github.com/twisted/twisted/issues/4486)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_4"></a>|@cyli set owner to @exarkun|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [29210]) Merge amp-key-identity-4486

Author: exarkun
Reviewer: cyli
Fixes: [#4486](https://github.com/twisted/twisted/issues/4486)

Weaken an assertion in one of the AMP header length tests: the identity
of the string containing the header name does not need to be preserved,
only the value.  This test failed on PyPy.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
